drivers/dma/dma_silabs_siwx917.c

```diff
@@ -0,0 +1,480 @@
+	/* Fetch the SRAM address where UDMA descriptor is stored */
+	UDMA_Table = (RSI_UDMA_DESC_T *)(cfg->reg->CTRL_BASE_PTR);
+
+	/* Check if the transfer type is memory-memory */
+	if ((UDMA_Table[rsi_channel].vsUDMAChaConfigData1.srcInc != UDMA_SRC_INC_NONE) &&
+	    (UDMA_Table[rsi_channel].vsUDMAChaConfigData1.dstInc != UDMA_DST_INC_NONE)) {
+		/* Apply software trigger to start transfer */
+		cfg->reg->CHNL_SW_REQUEST |= SET_BIT(rsi_channel);
+	}
+	return 0;
+}
+
+/* Function to stop a DMA transfer */
+static int dma_siwx917_stop(const struct device *dev, uint32_t channel)
+{
+	const struct dma_siwx917_config *cfg = dev->config;
+	struct dma_siwx917_data *data = dev->data;
+	RSI_UDMA_HANDLE_T udma_handle = (RSI_UDMA_HANDLE_T)data->dma_rom_buff;
+
+	/* Expecting a fixed channel number between 1-32 for udma0 and 1-12 for udma1 */
+	if ((channel > cfg->channels) || (channel == 0)) {
+		return -EINVAL;
+	}
+	/* Disable UDMA channel */
+	if (RSI_UDMA_ChannelDisable(udma_handle, (channel - 1)) != 0) {
+		return -EIO;
+	}
+	return 0;
+}
+
+/* Function to fetch DMA channel status */
+static int dma_siwx917_get_status(const struct device *dev, uint32_t channel,
+				  struct dma_status *stat)
+{
+	const struct dma_siwx917_config *cfg = dev->config;
+	uint32_t rsi_channel = channel - 1;
+	RSI_UDMA_DESC_T *UDMA_Table;
+
+	/* Expecting a fixed channel number between 1-32 for udma0 and 1-12 for udma1 */
+	if ((channel > cfg->channels) || (channel == 0)) {
+		return -EINVAL;
+	}
+	/* Read the channel status register */
+	if ((cfg->reg->CHANNEL_STATUS_REG >> rsi_channel) & 0x01) {
+		stat->busy = 1;
+	} else {
+		stat->busy = 0;
+	}
+	/* Fetch the SRAM address where UDMA descriptor is stored */
+	UDMA_Table = (RSI_UDMA_DESC_T *)(cfg->reg->CTRL_BASE_PTR);
+
+	/* Obtain the transfer direction from channel descriptors */
+	if (UDMA_Table[rsi_channel].vsUDMAChaConfigData1.srcInc == UDMA_SRC_INC_NONE) {
+		stat->dir = PERIPHERAL_TO_MEMORY;
+	} else if (UDMA_Table[rsi_channel].vsUDMAChaConfigData1.dstInc == UDMA_DST_INC_NONE) {
+		stat->dir = MEMORY_TO_PERIPHERAL;
+	} else {
+		stat->dir = MEMORY_TO_MEMORY;
+	}
+	return 0;
+}
+
+/* Function to initialize DMA peripheral */
+static int dma_siwx917_init(const struct device *dev)
+{
+	const struct dma_siwx917_config *cfg = dev->config;
+	struct dma_siwx917_data *data = dev->data;
+	RSI_UDMA_HANDLE_T udma_handle = NULL;
+	/* Initialize UDMA Resources */
+	UDMA_RESOURCES UDMA_Siwx917_Resources = {
+		.reg = cfg->reg, /* UDMA register base address */
+		.udma_irq_num = cfg->irq_number,
+	};
+
+	if (cfg->reg == UDMA0) {
+		UDMA_Siwx917_Resources.desc =
+			(RSI_UDMA_DESC_T *)DT_PROP(DT_NODELABEL(udma0), sram_desc_addr);
+	} else if (cfg->reg == UDMA1) {
+		UDMA_Siwx917_Resources.desc =
+			(RSI_UDMA_DESC_T *)DT_PROP(DT_NODELABEL(udma1), sram_desc_addr);
+	} else {
+		return -EINVAL;
+	}
+
+	udma_handle = UDMAx_Initialize(&UDMA_Siwx917_Resources, UDMA_Siwx917_Resources.desc,
+				       udma_handle, data->dma_rom_buff);
```

Comment:
What is purpose of passing `udma_handle` (which is guaranteed to be`== NULL`) to `UDMAx_Initialize()`?

Reply:
Agreed, passing NULL instead of udma_handle.